tests/e2e/loadbalancer.go

```diff
@@ -41,61 +70,334 @@ var _ = Describe("[cloud-provider-aws-e2e] loadbalancer", func() {
+			Name:           "NLB should configure the loadbalancer based on annotations",
+			ResourceSuffix: "nlb",
+			Annotations: map[string]string{
+				annotationLBType: "nlb",
+			},
+		},
+		{
+			Name:           "NLB should configure the loadbalancer with target-node-labels",
+			ResourceSuffix: "sg-nd",
+			Annotations: map[string]string{
+				annotationLBType: "nlb",
+			},
+			PostConfigService: func(cfg *configServiceLB, svc *v1.Service) {
+				// discover clusterNodeSelector and patch service
+				// TODO: move to external function if there are more scenarios to discover nodes.
+				By("discovering node label used in the kubernetes distributions")
+				for _, selector := range lookupNodeSelectors {
+					nodeList, err := cs.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+						LabelSelector: selector,
+					})
+					framework.ExpectNoError(err, "failed to list worker nodes")
+					if len(nodeList.Items) > 0 {
+						clusterNodesCount = len(nodeList.Items)
+						clusterNodesSelector = selector
+						break
+					}
+				}
+
+				if clusterNodesCount == 0 {
+					framework.ExpectNoError(fmt.Errorf("unable to find node selector for %v", lookupNodeSelectors))
+				}
+
+				By(fmt.Sprintf("found %d nodes with selector %q\n", clusterNodesCount, clusterNodesSelector))
+				if svc.Annotations == nil {
+					svc.Annotations = map[string]string{}
+				}
+				svc.Annotations[annotationLBTargetNodeLabels] = clusterNodesSelector
+				By(fmt.Sprintf("using service with annotations: %v", svc.Annotations))
+			},
+			PostRunValidation: func(cfg *configServiceLB, svc *v1.Service) {
+				// Validate in the TG if the node count matches with expected target-node-labels selector.
+				if len(svc.Status.LoadBalancer.Ingress) == 0 {
+					framework.Failf("No ingress found in LoadBalancer status for service %s/%s", svc.Namespace, svc.Name)
+				}
+				lbDNS := svc.Status.LoadBalancer.Ingress[0].Hostname
+				framework.ExpectNoError(getLBTargetCount(context.TODO(), lbDNS, clusterNodesCount), "AWS LB target count validation failed")
+			},
+		},
+	}
+
+	serviceNameBase := "lbconfig-test"
+	for _, tc := range cases {
+		It(tc.Name, func() {
+			loadBalancerCreateTimeout := e2eservice.GetServiceLoadBalancerCreationTimeout(cs)
+			framework.Logf("Running tests against AWS with timeout %s", loadBalancerCreateTimeout)
+
+			// Create Configuration
+			serviceName := serviceNameBase
+			if len(tc.ResourceSuffix) > 0 {
+				serviceName = serviceName + "-" + tc.ResourceSuffix
+			}
+			framework.Logf("namespace for load balancer conig test: %s", ns.Name)
+
+			By("creating a TCP service " + serviceName + " with type=LoadBalancerType in namespace " + ns.Name)
+			lbConfig := newConfigServiceLB()
+			lbConfig.LBJig = e2eservice.NewTestJig(cs, ns.Name, serviceName)
+			lbServiceConfig := lbConfig.buildService(tc.Annotations)
+
+			// Hook: PostConfigService patchs service configuration.
+			if tc.PostConfigService != nil {
+				By("running post run validations")
```

Comment:
i know this is a minor nit, but given that we have the same `By` statement on line 178, should we change this to say "running post configuration on service"?

Reply:
fair point, thanks for cathing that. We also have `By` on line 102, inside `PostConfigService` which would duplicate this one